### DISCUSSION TITLE
Search overlay: Insert on Enter

### DISCUSSION THREAD

**Discussion by MartyMcFlyInTheSky** [wezterm/wezterm#6293] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

i3

### WezTerm version

wezterm 20240915-153243-2d0c5cdd

### Ask your question!

From the bottom of the [search overlay help page:](https://wezfurlong.org/wezterm/scrollback.html#configuring-saved-searches)

> With that in your config you can now:
> 
>     CTRL-SHIFT-H to highlight all the git hashes and select the closest one to the bottom of the screen.
>     Use ENTER/CTRL-N/CTRL-P to cycle through the git hashes
>     CTRL-SHIFT-C to copy
>     Escape
>     CTRL-SHIFT-V (or SHIFT-Insert) to Paste
> 
> without needing to reach for your mouse.

I always found the last three steps a bit tedious shortcut-worthy: Is there a way I could configure wezterm such that it will automatically copy, escape the search overlay and insert (the last three steps) the text IF the foreground application is bash?

**Comment by bew** [comment, marked as answer]
You should try QuickSelect, it's does exactly that but with hints, when you type the hint in lowercase it copies the text, but when you type the hint in UPPERCASE, it directly inserts the text 😉
https://wezfurlong.org/wezterm/quickselect.html

**Reply by MartyMcFlyInTheSky** [reply]
Nice thanks for pointing that out. After this I also discovered CopyMode. Really powerful stuff!
